src/Text/Layout/Table/Cell/WideString.hs

```diff
@@ -0,0 +1,77 @@
+{-# LANGUAGE GeneralisedNewtypeDeriving #-}
+{-# LANGUAGE MultiWayIf                 #-}
+{-# LANGUAGE OverloadedStrings          #-}
+module Text.Layout.Table.Cell.WideString
+    ( WideString(..)
+    , WideText(..)
+    ) where
+
+import Data.String
+import qualified Data.Text as T
+import Text.DocLayout
+
+import Text.Layout.Table.Cell
+import Text.Layout.Table.Primitives.AlignInfo
+import Text.Layout.Table.StringBuilder
+
+-- | A newtype for String in which characters can be wider than one space.
+newtype WideString = WideString String
+    deriving (Eq, Ord, Show, Read, Semigroup, Monoid, IsString, StringBuilder)
```

Comment:
What would be the advantage of a `StringBuilder` instance?

Reply:
Probably nothing. There's a `StringBuilder` instance for `String` (and now `Text` as well), so I added these for symmetry. I'll remove.